bionic/persistence.py

```diff
@@ -1027,6 +983,18 @@ class Provenance:
     must be a versioning error.
     """
 
+    descriptor: str = attr.ib()
+    # We don't use this for anything; it's just included to allow humans to distinguish
```

Comment:
This comment applies only to `case_key_elements`, right? If so, can you add a newline before `dep_digests`? Otherwise, it seems like the comment applies to both the fields with the newlines separating other fields.

Reply:
Yep, good point!